src/sqlfluff/rules/structure/ST10.py

```diff
@@ -0,0 +1,71 @@
+"""Implementation of Rule ST10."""
+
+from typing import Iterator, Tuple
+
+from sqlfluff.core.rules import BaseRule, EvalResultType, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_ST10(BaseRule):
+    """Const expression rule.
+
+    **Anti-pattern**
+
+    .. code-block:: sql
+
+         -- TODO
+        SELECT 1;
+
+    **Best practice**
+
+    .. code-block:: sql
+
+         -- TODO
+        SELECT 1;
+
+    """
```

Comment:
```suggestion
    """Redundant constant condition.
    
    Including a condition in a ``WHERE`` clause which always evaluates to
    either ``TRUE`` or ``FALSE`` regardless of the input columns is
    unnecessary and makes statements harder to read and understand.
    
    Constant conditions are sometimes mistakes (by mistyping the column
    name intended), and sometimes the result of incorrect information that
    they are necessary in some circumstances. Given this ambiguity, this
    rule does not suggest an automatic fix, and instead invites the user
    to resolve the problem manually.

    **Anti-pattern**

    .. code-block:: sql

        SELECT *
        FROM my_table
        -- This following WHERE clause is redundant.
        WHERE my_table.col = my_table.co1

    **Best practice**

    .. code-block:: sql

        SELECT *
        FROM my_table
        -- Replace with a condition that includes meaningful logic,
        -- or remove the condition entirely.
        WHERE my_table.col > 3 

    """
```

Reply:
I'll apply these docs as starting point, but here as well - we have constant_expression, not constant condition.